src/openloco/windows/construction/Common.cpp

```diff
@@ -0,0 +1,2036 @@
+        }
+
+        auto window = WindowManager::find(WindowType::construction);
+
+        if (window != nullptr)
+        {
+            if (flags & (1 << 7))
+            {
+                auto trackType = flags & ~(1 << 7);
+                auto roadObj = objectmgr::get<road_object>(trackType);
+
+                if (roadObj->flags & flags_12::unk_03)
+                {
+                    if (_trackType & (1 << 7))
+                    {
+                        trackType = _trackType & ~(1 << 7);
+                        roadObj = objectmgr::get<road_object>(trackType);
+
+                        if (roadObj->flags & flags_12::unk_03)
+                        {
+                            _trackType = flags;
+
+                            common::sub_4A3A50();
+
+                            _lastSelectedTrackPiece = 0;
+                            _lastSelectedTrackGradient = 0;
+
+                            return window;
+                        }
+                    }
+                }
+            }
+        }
+
+        WindowManager::closeConstructionWindows();
+        common::sub_4CD454();
+
+        mainWindow = WindowManager::getMainWindow();
+
+        if (mainWindow)
+        {
+            auto viewport = mainWindow->viewports[0];
+            viewport->flags = _word_1135F86;
+        }
+
+        _trackType = static_cast<uint8_t>(flags);
+        _byte_1136063 = flags >> 24;
+        _x = 0x1800;
+        _y = 0x1800;
+        _word_1135FB8 = 0x100;
+        _constructionRotation = 0;
+        _constructionHover = 0;
+        _byte_113607E = 1;
+        _trackCost = 0x80000000;
+        _byte_1136076 = 0;
+        _lastSelectedTrackPiece = 0;
+        _lastSelectedTrackGradient = 0;
+        _lastSelectedTrackModSection = 0;
+
+        common::setTrackOptions(flags);
+
+        if (flags & (1 << 31))
+        {
+            return createAirportConstructionWindow();
+        }
+        else if (flags & (1 << 30))
+        {
+            return createDockConstructionWindow();
+        }
+        else if (flags & (1 << 7))
+        {
+            return createRoadConstructionWindow();
+        }
+
+        return createTrackConstructionWindow();
+    }
+
+    // 0x004A6FAC
+    void sub_4A6FAC()
+    {
+        auto window = WindowManager::find(WindowType::construction);
+        if (window == nullptr)
+            return;
+        if (window->current_tab == common::widx::tab_station - common::widx::tab_construction)
+        {
+            if (_byte_1136063 & ((1 << 7) | (1 << 6)))
+                WindowManager::close(window);
+            else
+                window->call_on_mouse_up(common::widx::tab_construction);
+        }
+    }
+
+    namespace common
+    {
+        struct TabInformation
+        {
+            widget_t* widgets;
+            const widx widgetIndex;
+            window_event_list* events;
+            const uint64_t enabledWidgets;
+            void tabReset(window* self)
```

Comment:
Sorry forgot to get around to saying how to do this. You can have a function pointer here. So you need a void function that passes a window argument. The would be like the following:

```
void (*tabReset)(window*);
```
You can then use that in your arrays below:
`{ construction::widgets, widx::tab_construction, &construction::events, construction::enabledWidgets, &construction::tabReset }`